simp-cli.gemspec

```diff
@@ -43,8 +43,11 @@ Gem::Specification.new do |s|
   ### When the environment variable $SIMP_RPM_BUILD is set, the following
   ### dependencies will not be included in the .gemspec dependencies:
   #### ------------
-  if !ENV.fetch( 'SIMP_RPM_BUILD', false )
-    s.add_runtime_dependency 'puppet',   '~> 4'
+  unless (
+    ENV.fetch( 'SIMP_RPM_BUILD', false ) || \
+    ENV.fetch( 'SIMP_CLI_GEMSPEC_NO_PUPPET_VERSION', false ) =~  /yes|true/
```

Comment:
I would like to see this explained somewhere for a developer who is trying to execute tests.  For example, in my dev environment which is Puppet 4, I can't run the tests unless I explicitly set PUPPET_VERSION="~> 4.10" SIMP_CLI_GEMSPEC_NO_PUPPET_VERSION=yes

Reply:
@lnemsick-simp This change was made so Travis pipelines could test the logic by using both Puppet versions—until now, the pipelines only used Puppet 4, even when testing ruby 2.4.4.

A new developer who is trying to execute tests would receive Puppet `~> 5` by running `bundle`.   If you have an existing Gemfile.lock with conflicting dependencies, run `bundle update` (as usual) to get the updated dependencies, unless you have a reason to stick with Puppet 4.

Based on this feedback, I've bulked up the README to explain this usage, and the things that need to be explained in order to make sense of that explanation.  In retrospect, it probably wasn't worth scope-creeping a time-sensitive quick patch with that, since (until now) the project hasn't even documented using `bundler`.